### PR TITLE
Made configuration optional by using archive configuration as default

### PR DESCRIPTION
**Pull request by mateherber**
If the configuration input is not set we select the scheme's archive
action's configuration.

[PLAT-51]

[PLAT-51]: https://bitrise.atlassian.net/browse/PLAT-51

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -183,6 +183,25 @@ func main() {
...
+		s, sc, err = readScheme(absProjectPath, cfg.Scheme)
+
+		if err != nil {
+			failf("Failed to read schema: %s", err)
+		}
+
+		if cfg.Configuration != "" {
+			conf = cfg.Configuration
+		} else {
+			conf = s.ArchiveAction.BuildConfiguration
```
Comment (lpusok) [inline main.go:201]: The Xcode UI uses the Configuration of the Run action when Building.

Comment (mateherber) [reply main.go:201]: Yes I experienced that the "LaunchAction"'s configuration is used by default if we don't feed anything to the xcodebuild. But for some reason we already have a fallback to the archive configuration at `findBuiltProject`. Also we don't have the launchaction included in the `xscheme.go` If we wish to default to that we need to update the `xcode-project`

File: main.go [hunk 2]
```diff
@@ -183,6 +183,25 @@ func main() {
 		failf("Failed to get absolute project path: %s", err)
 	}
 
+	//
+	// Read the Scheme
+	var s *xcscheme.Scheme
+	var sc string
+	var conf string
```
Comment (Szabadember) [inline main.go:190]: Shouldn't we use more descriptive variable names here?

**Review by Szabadember — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (lpusok) [reply main.go]: The current solution is acceptable, please document it in the step.yml Configuration input description.

Comment (mateherber) [reply main.go]: Updated the description

**Review by lpusok — APPROVED** [review]

**Review by Szabadember — APPROVED** [review]

**Pull request merged**